sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala

```diff
@@ -699,8 +701,42 @@ case class UnionExec(children: Seq[SparkPlan]) extends SparkPlan {
     }
   }
 
+  private lazy val childrenRDDs = children.map(_.execute())
```

Comment:
Actaully, does this mean that children executions are triggered to get `outputPartitioning` of an `Union`?
E.g. a simple explain to show the physical plan now can trigger execuion of union children?

Reply:
Yes, so this approach has this drawback. So as I mentioned https://github.com/apache/spark/pull/51623#discussion_r2232350704, this doesn't cover SQL cases generally. I plan to extend this to deal with `outputPartitioning` of children, i.e., no need to invoke `execute` on children.

It will be done in follow up works.

Reply:
I actually have the PR ready locally. After this gets merged, I will open a new PR to improve it and get rid of this `execute` calls.

Reply:
IMO that's a serious drawback. But if we can fix it in a follow-up PR right after this PR then I'm ok with merging. Or just update this PR with you local changes.

Reply:
Okay, I updated to this PR.

Reply:
Thanks. I can check it tomorrow.